File: resources/expected/ros2/isolette/lax/src/isolette_cpp_pkg/src/user_code/thermostat_monitor_temperature_manage_alarm_mat_src.cpp
#include "isolette_cpp_pkg/user_headers/thermostat_monitor_temperature_manage_alarm_mat_src.hpp"
#include "isolette_cpp_pkg/base_headers/enum_converter.hpp"

//===========================================================
// This file will not be overwritten when re-running Codegen
//===========================================================

//=================================================
//  I n i t i a l i z e    E n t r y    P o i n t
//=================================================
void thermostat_monitor_temperature_manage_alarm_mat::initialize()
{
    PRINT_INFO("Initialize Entry Point invoked");

    // Initialize the node
    // Initialize the node's incoming data port values here
    isolette_cpp_pkg_interfaces::msg::TempWstatusimpl current_tempWstatus = isolette_cpp_pkg_interfaces::msg::TempWstatusimpl();
    init_current_tempWstatus(current_tempWstatus);

    isolette_cpp_pkg_interfaces::msg::Tempimpl lower_alarm_temp = isolette_cpp_pkg_interfaces::msg::Tempimpl();
    init_lower_alarm_temp(lower_alarm_temp);

    isolette_cpp_pkg_interfaces::msg::Tempimpl upper_alarm_temp = isolette_cpp_pkg_interfaces::msg::Tempimpl();
    init_upper_alarm_temp(upper_alarm_temp);

    isolette_cpp_pkg_interfaces::msg::MonitorMode monitor_mode = isolette_cpp_pkg_interfaces::msg::MonitorMode();
    init_monitor_mode(monitor_mode);

}

//=================================================
//  C o m p u t e    E n t r y    P o i n t
//=================================================
void thermostat_monitor_temperature_manage_alarm_mat::timeTriggered()
{
    // Handle communication

    // example receiving messages on data ports
    isolette_cpp_pkg_interfaces::msg::TempWstatusimpl::SharedPtr current_tempWstatus = get_current_tempWstatus();
    PRINT_INFO("Received current_tempWstatus: %s", MESSAGE_TO_STRING(current_tempWstatus));
    isolette_cpp_pkg_interfaces::msg::Tempimpl::SharedPtr lower_alarm_temp = get_lower_alarm_temp();
    PRINT_INFO("Received lower_alarm_temp: %s", MESSAGE_TO_STRING(lower_alarm_temp));
    isolette_cpp_pkg_interfaces::msg::Tempimpl::SharedPtr upper_alarm_temp = get_upper_alarm_temp();
    PRINT_INFO("Received upper_alarm_temp: %s", MESSAGE_TO_STRING(upper_alarm_temp));
    isolette_cpp_pkg_interfaces::msg::MonitorMode::SharedPtr monitor_mode = get_monitor_mode();
    PRINT_INFO("Received monitor_mode: %s", MESSAGE_TO_STRING(monitor_mode));

    // Example publishing messages
    isolette_cpp_pkg_interfaces::msg::OnOff alarm_control = isolette_cpp_pkg_interfaces::msg::OnOff();
    put_alarm_control(alarm_control);
}

File: resources/expected/ros2/python-isolette/strict/src/isolette_py_pkg/isolette_py_pkg/user_code/thermostat_monitor_temperature_manage_alarm_mat_src.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from rosidl_runtime_py.convert import message_to_yaml
from isolette_py_pkg.base_code.thermostat_monitor_temperature_manage_alarm_mat_base_src import thermostat_monitor_temperature_manage_alarm_mat_base
from isolette_py_pkg_interfaces.msg import TempWstatusimpl
from isolette_py_pkg_interfaces.msg import Tempimpl
from isolette_py_pkg_interfaces.msg import MonitorMode
from isolette_py_pkg_interfaces.msg import OnOff
from isolette_py_pkg.base_code.enum_converter import *

#===========================================================
# This file will not be overwritten when re-running Codegen
#===========================================================
class thermostat_monitor_temperature_manage_alarm_mat(thermostat_monitor_temperature_manage_alarm_mat_base):
    def __init__(self):
        super().__init__()
        # invoke initialize entry point
        self.initialize()

        self.get_logger().info("thermostat_monitor_temperature_manage_alarm_mat infrastructure set up")

#=================================================
#  I n i t i a l i z e    E n t r y    P o i n t
#=================================================
    def initialize(self):
        self.get_logger().info("Initialize Entry Point invoked")

        # Initialize the node

        # Initialize the node's incoming data port values here

        # Initialize the node's incoming data port values here
        current_tempWstatus = TempWstatusimpl()
        self.init_current_tempWstatus(current_tempWstatus)

        lower_alarm_temp = Tempimpl()
        self.init_lower_alarm_temp(lower_alarm_temp)

        upper_alarm_temp = Tempimpl()
        self.init_upper_alarm_temp(upper_alarm_temp)

        monitor_mode = MonitorMode()
        self.init_monitor_mode(monitor_mode)


#=================================================
#  C o m p u t e    E n t r y    P o i n t
#=================================================
    def message_to_string(self, msg):
        yaml_str = message_to_yaml(msg)
        return yaml_str

    def timeTriggered(self):
        # Handle communication

        # Example receiving messages on data ports
        current_tempWstatus = self.get_current_tempWstatus()
        self.get_logger().info(f"Received current_tempWstatus: {self.message_to_string(current_tempWstatus)}")

        lower_alarm_temp = self.get_lower_alarm_temp()
        self.get_logger().info(f"Received lower_alarm_temp: {self.message_to_string(lower_alarm_temp)}")

        upper_alarm_temp = self.get_upper_alarm_temp()
        self.get_logger().info(f"Received upper_alarm_temp: {self.message_to_string(upper_alarm_temp)}")

        monitor_mode = self.get_monitor_mode()
        self.get_logger().info(f"Received monitor_mode: {self.message_to_string(monitor_mode)}")


        # Example publishing messages
        alarm_control = OnOff()
        self.put_alarm_control(alarm_control)


#=================================================
#  Include any additional declarations here
#=================================================
# Additions within these tags will be preserved when re-running Codegen

# Additions within these tags will be preserved when re-running Codegen
